Classes/Task/BaseAdditionalFieldProvider.php

```diff
@@ -0,0 +1,281 @@
+
+        $messageTitle = Helper::getLanguageService()->getLL('additionalFields.error');
+        $messageSeverity = FlashMessage::ERROR;
+
+        if (isset($submittedData['doc']) && empty($submittedData['doc'])) {
+            Helper::addMessage(
+                Helper::getLanguageService()->getLL('additionalFields.doc') . ' ' . Helper::getLanguageService()->getLL('additionalFields.valid'),
+                $messageTitle,
+                $messageSeverity,
+                true,
+                'core.template.flashMessages'
+            );
+            $fieldsValid = false;
+        }
+
+        if ((isset($submittedData['pid']) && (int) $submittedData['pid'] <= 0) || !isset($submittedData['pid'])) {
+            Helper::addMessage(
+                Helper::getLanguageService()->getLL('additionalFields.pid') . ' ' . Helper::getLanguageService()->getLL('additionalFields.valid'),
+                $messageTitle,
+                $messageSeverity,
+                true,
+                'core.template.flashMessages'
+            );
+            $fieldsValid = false;
+        }
+
+        if (!$submittedData['uid']) {
+            $messageTitle = Helper::getLanguageService()->getLL('additionalFields.warning');
+            $messageSeverity = FlashMessage::WARNING;
+        }
+
+        if ((isset($submittedData['lib']) && (int) $submittedData['lib'] <= 0)) {
+            Helper::addMessage(
+                Helper::getLanguageService()->getLL('additionalFields.lib') . ' ' . Helper::getLanguageService()->getLL('additionalFields.valid'),
+                $messageTitle,
+                $messageSeverity,
+                true,
+                'core.template.flashMessages'
+            );
+            $fieldsValid = false;
+        }
+
+        if ((isset($submittedData['solr']) && (int) $submittedData['solr'] <= 0) || !isset($submittedData['solr'])) {
+            Helper::addMessage(
+                Helper::getLanguageService()->getLL('additionalFields.solr') . ' ' . Helper::getLanguageService()->getLL('additionalFields.valid'),
+                $messageTitle,
+                $messageSeverity,
+                true,
+                'core.template.flashMessages'
+            );
+            $fieldsValid = false;
+        }
+
+        if (((isset($submittedData['coll']) && isset($submittedData['all'])) || (!isset($submittedData['coll']) && !isset($submittedData['all'])))
+            && !isset($submittedData['doc']) && !isset($submittedData['lib'])) {
+            Helper::addMessage(
+                Helper::getLanguageService()->getLL('additionalFields.collOrAll'),
+                $messageTitle,
+                $messageSeverity,
+                true,
+                'core.template.flashMessages'
+            );
+            $fieldsValid = false;
+        }
+        return $fieldsValid;
+    }
+
+    public function saveAdditionalFields(array $submittedData, AbstractTask $task)
```

Comment:
```suggestion
    public function saveAdditionalFields(array $submittedData, BaseTask $task)
```
PHPStan correctly mentions all the calls to undefined methods in this method. That's because you declare `$task` as being of type `AbstractTask`, but then call methods which are only defined in `BaseTask`. I think declaring `$task` as being of type `BaseTask` should fix this. (On the other hand this may break the interface compatibility...)

Reply:
Alternatively, you can add a comment for PHPStan that clarifies the type of `$task` as being `BaseTask`. The comment should be placed directly above the first usage of `$task`:
```php
/* @var BaseTask $task */
```